Physics/CC-33-Radiation-Animations/Beta_minus/Beta_minus.pde

```diff
@@ -0,0 +1,234 @@
+class TextBox {
+  PVector location;
+  PVector velocity;
+  
+  TextBox(float x_loc, float y_loc, float x_vel, float y_vel) {
+    location = new PVector(x_loc, y_loc);
+    velocity = new PVector(x_vel, y_vel);
+  }
+  void update() {
+    location.add(velocity);
+  }
+  
+  void display(String message, int size, float[] col) {
+    textSize(size);
+    fill(col[0],col[1],col[2]);
+    text(message, location.x, location.y);
+  }
+}
+
+
+
+//for beta- decay, releases electron and anti neutrino
+//large number of neutrinos, neurtino changes into a proton
+
+class Mover {
+  PVector location;
+  PVector velocity;
+  PVector acceleration;
+  
+  
+  Mover(float x_loc,float y_loc, float x_vel, float y_vel) {
+    
+    location = new PVector(x_loc,y_loc);
+    velocity = new PVector(x_vel,y_vel);
+    
+  }
+  
+  void update() {
+    location.add(velocity);
+    
+  }
+    
+  void display(int i, int rad, int col) {
+    stroke(0);
+    //red
+    if (col == 0) {
+      fill(255,0,0);
+    }
+    //green
+    else if(col == 1) {
+      fill(35, 167, 6);
+    }
+    //blue
+    else if(((i)%3)==0) {
+      fill(12,48,224);
+    } //grey
+    else {
+      fill(175);
+    }
+    ellipse(location.x,location.y,rad,rad);
+  }
+}
+
+
+// new function
+float[][] position(float x_val, float y_val, float radius) {
+  // make it produce the x and y coordinates for the 6 spots around it
+  
+  float y_dist = (sin(radians(30)))*(2*radius);
+  float x_dist = (cos(radians(30)))*(2*radius);
+  
+  //coords is a list that contains the new coordinates for 6 circles around one circle
+  
+  
+  float[][] coords = {{(x_val),(y_val-(2*radius))}, {(x_val+x_dist),(y_val-y_dist)}, {(x_val+x_dist),(y_val+y_dist)}, {(x_val),(y_val+(2*radius))}, {(x_val-x_dist),(y_val+y_dist)}, {(x_val-x_dist),(y_val-y_dist)}};
+  
+  return coords;
+}
+
+Mover[] movers = {};
+
+
+//function to check if a coordinate is already used
+boolean isItemInArray(float[][] array, float[] item) {
```

Comment:
Functions like these which are being used at multiple places can be moved to a separate helper class and referenced from there.